From 57ec67c56ea05480eef2df730cef8b0dc790d8ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@redhat.com>
Date: Thu, 16 Oct 2014 11:19:05 +0200
Subject: [PATCH] Remove call to CrtcRotate()

Our driver doesn't support rotation, and calling CrtcRotoate()
may result in crash. Let's remove that useless call.

Fixes:
http://bugzilla.redhat.com/1067709
---
 src/qxl_drmmode.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/qxl_drmmode.c b/src/qxl_drmmode.c
index 42347e6..35d7897 100644
--- a/src/qxl_drmmode.c
+++ b/src/qxl_drmmode.c
@@ -197,9 +197,6 @@ drmmode_set_mode_major(xf86CrtcPtr crtc, DisplayModePtr mode,
 			output_count++;
 		}
 
-		if (!xf86CrtcRotate(crtc)) {
-			goto done;
-		}
 #if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,7,0,0,0)
 		crtc->funcs->gamma_set(crtc, crtc->gamma_red, crtc->gamma_green,
 				       crtc->gamma_blue, crtc->gamma_size);
@@ -354,6 +351,7 @@ drmmode_crtc_init(ScrnInfoPtr pScrn, drmmode_ptr drmmode, int num)
 	crtc = xf86CrtcCreate(pScrn, &drmmode_crtc_funcs);
 	if (crtc == NULL)
 		return;
+	crtc->driverIsPerformingTransform = FALSE;
 
 	drmmode_crtc = xnfcalloc(sizeof(drmmode_crtc_private_rec), 1);
 	drmmode_crtc->mode_crtc = drmModeGetCrtc(drmmode->fd, drmmode->mode_res->crtcs[num]);
